geoh5py/shared/utils.py

```diff
@@ -382,6 +382,34 @@ def dict_mapper(
     return val
 
 
+def box_intersect(
+    extent_a: np.ndarray, extent_b: np.ndarray, return_array=False
+) -> bool | np.ndarray:
+    """
+    Compute the intersection of two axis-aligned bounding extents defined by their
+    bottom south-west corner and top north-east corner.
+
+    :param extent_a: First extent
```

Comment:
The `zip()` will stop at the lowest common number. So if one is (2,2) and the other (2,3), then only the x and y will be checked. Makes sense?

Reply:
that is correct if just testing the intersection, but returns the wrong bounding box if computing the resulting intersection box.

Reply:
not relevant since not returning the box anymore, but just to explain my comment:
with the code in place, computing the intersection between a 2D box and a 3D box would have created a 2D box, while it must preserve the limits on z and thus returns a 3D box